masq/src/non_interactive_mode.rs

```diff
@@ -122,71 +109,69 @@ impl command::Command for Main {
                         "Can't connect to Daemon or Node ({:?}). Probably this means the Daemon isn't running.",
                         error
                     );
-                return 1;
+                return bool_into_numeric_code(false);
             }
         };
 
         let result = match subcommand_opt {
-            Some(command_parts) => {
-                match handle_command_common(
-                    &*self.command_factory,
-                    &mut *command_processor,
-                    command_parts,
-                    streams.stderr,
-                ) {
-                    Ok(_) => 0,
-                    Err(_) => 1,
-                }
-            }
-            None => go_interactive(
-                Box::new(handle_command_common),
+            Some(command_parts) => handle_command_common(
                 &*self.command_factory,
                 &mut *command_processor,
-                streams,
+                command_parts,
+                streams.stderr,
             ),
+            None => go_interactive(&*self.command_factory, &mut *command_processor, streams),
         };
         command_processor.close();
-        result
+        bool_into_numeric_code(result)
     }
 }
 
-fn handle_command_common(
-    command_factory: &(dyn CommandFactory + 'static),
-    processor: &mut (dyn CommandProcessor + 'static),
+fn bool_into_numeric_code(bool_flag: bool) -> u8 {
+    bool_flag.not() as u8
```

Comment:
I'd rather have an if statement here, even if it's longer, because it'd be clearer what was going on.